src/components/button/button.tsx

```diff
@@ -80,6 +80,7 @@ export interface EuiButtonProps extends CommonProps {
    * Object of props passed to the <span/> wrapping the component's {children}
    */
   textProps?: HTMLAttributes<HTMLSpanElement>;
+  onClick?: (prop?: any) => void;
```

Comment:
@myasonik can you help me in assigning a more strict type to this

Reply:
I didn't test this but I believe this is correct:

```suggestion
  onClick?: MouseEventHandler<HTMLButtonElement>
```